src/FilesystemManagement.cpp

```diff
@@ -0,0 +1,365 @@
+   out:
+    - DRM or Copy Protection of any kind(except Copyrights)
+    - Software Patents You Do Not Wish to Freely License
+    - Any Kind of Linking to Non-GPL licensed Works
+    - Are Currently In Violation of Another Copyright Holder's GPL License
+    - If You want to change our code and not add a few hundred MB of stuff to
+        your distribution
+
+   These and other limitations could cause serious legal problems if you ignore
+   them, so it is best to simply contact us or the Free Software Foundation, if
+   you have any questions.
+
+   Joseph Toppi - toppij@gmail.com
+   John Blackwood - makoenergy02@gmail.com
+*/
+
+#include "CrossPlatformExport.h"
+
+#ifdef MEZZ_Windows
+    #define _WIN32_WINNT 0x0601
+#endif
+
+#include "FilesystemManagement.h"
+#include "PathUtilities.h"
+#include "StringTools.h"
+
+#include <cstring>
+#include <iostream>
+
+#ifdef MEZZ_Windows
+    #define WIN32_LEAN_AND_MEAN
+
+    SAVE_WARNING_STATE
+    SUPPRESS_VC_WARNING(4668)
+
+    #include <Windows.h>
+
+    RESTORE_WARNING_STATE
+#else
+    #include <stdio.h>
+    #include <sys/stat.h>
+    #include <sys/types.h>
+    #include <unistd.h>
+#endif
+
+#include "PlatformUndefs.h"
+
+namespace
+{
+    using namespace Mezzanine;
+
+#ifdef MEZZ_Windows
+    /// @brief Converts a narrow (8-bit) string to a wide (16-bit) string.
+    /// @param Narrow The string to be converted.
+    /// @return Returns a wide string with the converted contents.
+    std::wstring ConvertToWideString(const StringView Narrow)
+    {
+        std::wstring Ret;
+        if( !Narrow.empty() ) {
+            int NarrowSize = static_cast<int>(Narrow.size());
+            size_t WideLength = ::MultiByteToWideChar(CP_UTF8,0,Narrow.data(),NarrowSize,0,0);
+            Ret.resize(WideLength,L'\0');
+            ::MultiByteToWideChar(CP_UTF8,0,Narrow.data(),NarrowSize,&Ret[0],static_cast<int>(Ret.size()));
+        }
+        return Ret;
+    }
+
+    /// @brief Converts the system error number to a Mezzanine ModifyResult.
+    /// @param err The system error to be converted.
+    /// @return Returns a ModifyResult value corresponding to the system error code.
+    Filesystem::ModifyResult ConvertErrNo(DWORD err)
+    {
+        switch( err )
+        {
+            case ERROR_SUCCESS:            return Filesystem::ModifyResult::Success;
+            case ERROR_ALREADY_EXISTS:     return Filesystem::ModifyResult::AlreadyExists;
+            case ERROR_FILE_EXISTS:        return Filesystem::ModifyResult::AlreadyExists;
+            case ERROR_FILE_NOT_FOUND:     return Filesystem::ModifyResult::DoesNotExist;
+            case ERROR_PATH_NOT_FOUND:     return Filesystem::ModifyResult::DoesNotExist;
+            case ERROR_INVALID_NAME:       return Filesystem::ModifyResult::InvalidPath;
+            case ERROR_BAD_PATHNAME:       return Filesystem::ModifyResult::InvalidPath;
+            //case :         return Filesystem::ModifyResult::LoopingPath;
```

Comment:
What are these commented out chunks? Should they be documented and removed or re-added?

Reply:
They look like things present in posix OSs, but not windows? if so a comment just saying that windows doesn't support them seems prudent.

Reply:
They've been removed.